Milestone3/logic

YoseptF opened the pull request:
We added the actions:
- REMOVE_BOOK to the bookm component
- ADD_BOOK to the form component

YoseptF committed:
We added the first files for the setup, and now we're ready to move ot milestone 2

YoseptF committed:
We implemented the compoents:
- App
- Bookslist
- Book
- Boosform

HADDADSOHAIB committed:
add book reducer and actions

HADDADSOHAIB committed:
create store, reducer and connect them

YoseptF committed:
We made the basic structure of the app, now we have:
- Book
- Booklist
We also co9nected to a store created in redux

YoseptF committed:
We added the actions:
- REMOVE_BOOK to the bookm component
- ADD_BOOK to the form component

YoseptF committed:
We added the filter in the redux store and also, created a component to filter these categories
Then we connected the components to the redux store.

YoseptF committed:
Revert "We added the filter in the redux store and also, created a component to filter these categories"

This reverts commit dc7f1b396758f5b884f5d5e3e629bb5a6cfcef52.

HADDADSOHAIB committed:
refactor the code

HADDADSOHAIB committed:
Merge branch 'milestone2/structure' of https://github.com/YoseptF/react-bookstore into milestone2/structure

HADDADSOHAIB committed:
add condition book reducer

HADDADSOHAIB committed:
Merge branch 'milestone2/structure' of https://github.com/YoseptF/react-bookstore into milestone3/logic

HADDADSOHAIB committed:
refacator the code

shloch commented on src/containers/BooksForm.js:
```diff
@@ -0,0 +1,71 @@
+      category: e.target.value,
+    });
+  }
+
```
### [optional] - optimization 

These two can be knocked into one function `handleChange` like ths :
```
handleChange(event) {
  const name = event.target.name;

  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}

```

shloch commented on src/containers/BooksForm.js:
```diff
@@ -0,0 +1,71 @@
+    });
+  }
+
+  handleSubmit() {
+    const { title, category } = this.state;
+    const { dispatch } = this.props;
+    dispatch(createBook({
+      id: Math.floor(Math.random() * 100),
+      title,
+      category,
+    }));
+  }
```
### [required]
You need to attach this function to the Form with `onSubmit` either on `<form>` tag or with a SUBMIT button.
Also, without 
```
`event.preventDefault();`
```
this might cause a page refresh which you dont want

shloch replied:
And then, at the end of this function (inside the function), you need to reset the state values [see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)]

```

this.setState({
    title: '',
    category: 'Action'
  });

```

shloch reviewed (changes requested):
## Hello @YoseptF , @HADDADSOHAIB 
Great job........
We're advancing quite nicely....please check comments below on the things to improve to meet the requirements completely before we can move on.

Great job on what you've done so far
### Happy coding

YoseptF committed:
We added a new way to handle change con the booksform, now it only needs one function
Also we added a button to submit new books and a way to prevent it from reloading the page :d

MauricioRobayo commented on src/containers/BookList.js:
```diff
@@ -0,0 +1,45 @@
+const BookList = ({ books, dispatch }) => {
+  const handleRemoveBook = e => {
+    const { id } = e.currentTarget.dataset;
+    dispatch(removeBook(id));
```
[OPTIONAL]: You can also use [`mapDispatchToProps`](https://react-redux.js.org/using-react-redux/connect-mapdispatch)

MauricioRobayo commented on src/containers/BooksForm.js:
```diff
@@ -0,0 +1,75 @@
+    const { name } = event.target;
+    const { state } = this;
+
+    this.setState({
+      ...state,
+      [name]: event.target.value,
+    });
```
👍 , nice. Just take into account that state updates are automatically merged by `setState`, so no need to do it yourselves.

https://reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged

MauricioRobayo reviewed (approved):
Great, looking good 👍 .

Thanks for fixing the issues mentioned before and improving the code.

Status: __Approved__.
